src/plays/image-compressor/ImageCompressor.js

```diff
@@ -0,0 +1,88 @@
+import { useState } from "react";
+import PlayHeader from "common/playlists/PlayHeader";
+import imageCompression from "browser-image-compression";
+import CircularProgress from "@mui/material/CircularProgress";
+import "./styles.css";
+
+function ImageCompressor(props) {
+  // Your Code Start below.
+
+  const [selectedFile, setSelectedFile] = useState(null);
+  const [selectedFileSize, setSelectedFileSize] = useState(null);
+  const [compressing, setCompressing] = useState(false);
+  const [compressedFile, setCompressedFile] = useState(null);
+  const [compressedFileSize, setCompressedFileSize] = useState(null);
+
+  const handleImageUpload = async (event) => {
+    const imageFile = event.target.files[0];
+    if (!imageFile) {
+      return;
+    }
+    setSelectedFile(imageFile);
+    setSelectedFileSize((imageFile.size / 1024 / 1024).toFixed(2));
+    setCompressing(true);
+    const options = {
+      maxSizeMB: 1,
+      maxWidthOrHeight: 1920,
+      useWebWorker: true,
+    };
+    try {
+      const compressedFile = await imageCompression(imageFile, options);
+      setCompressedFile(compressedFile);
+      setCompressedFileSize((compressedFile.size / 1024 / 1024).toFixed(2));
+      setCompressing(false);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  return (
+    <>
+      <div className="play-details">
+        <PlayHeader play={props} />
+        <div className="play-details-body">
+          {/* Your Code Starts Here */}
+          <div className="imageCompressor">
+            <input
+              className="imageCompressor__input"
+              type="file"
+              accept="image/*"
+              onChange={handleImageUpload}
+            />
+            <div className="imageCompressor__imageContainer">
+              <div className="imageCompressor__box">
+                <div className="imageCompressor__image">
+                  {selectedFile && (
+                    <img src={URL.createObjectURL(selectedFile)} alt="" />
+                  )}
+                </div>
+                <p>{selectedFileSize} MB</p>
+              </div>
+              <div className="imageCompressor__progress">
+                {compressing && <CircularProgress />}
+              </div>
+              <div className="imageCompressor__box">
+                <div className="imageCompressor__image">
+                  {compressedFile && (
+                    <img src={URL.createObjectURL(compressedFile)} alt="" />
+                  )}
+                </div>
+                <p>{compressedFileSize} MB</p>
+              </div>
+            </div>
+            {compressedFile && (
```

Comment:
We have the same check above `compressedFile &&`, can we refactor the code and use this check one time and render both the compressed image and download button?

Reply:
Both the checks are in different div. I think its impossible to render both using one check.

Reply:
@aks318 , Can't we use a fragment here and render both div with one condition?

Example:

```js
{condition && <>
<div1></div1>
<div2></div2>
</> }

```

Reply:
![image](https://user-images.githubusercontent.com/41566374/199082301-be16eab4-cce7-49a7-8435-05d474963746.png)

@Sachin-chaurasiya , I could have done it if they are siblings. But according to UI, 1st check renders image and 2nd renders button.

Reply:
@aks318 , Both depend on the same condition, right?
Can you try the suggested change?

Reply:
@aks318 , no worries, I got the point about why we can't do that.

Reply:
@Sachin-chaurasiya Just opened my laptop and saw your comment. Great to see you got my point.